jobs/build.Jenkinsfile

```diff
@@ -634,65 +640,68 @@ lock(resource: "build-${params.STREAM}") {
-            parallelruns['Kola:AWS'] = {
-                // We consider the AWS kola tests to be a followup job, so we use `wait: false` here.
-                build job: 'kola-aws', wait: false, parameters: [
-                    string(name: 'STREAM', value: params.STREAM),
-                    string(name: 'VERSION', value: newBuildID),
-                    string(name: 'S3_STREAM_DIR', value: s3_stream_dir),
-                    string(name: 'ARCH', value: basearch),
-                    string(name: 'SRC_CONFIG_COMMIT', value: src_config_commit)
-                ]
+        if (!params.MINIMAL && uploading) {
+            // Kick off the Kola AWS job if we have credentials for running those tests.
+            pipeutils.tryWithCredentials([file(variable: 'AWS_KOLA_TESTS_CONFIG',
+                                               credentialsId: 'aws-kola-tests-config')]) {
+                parallelruns['Kola:AWS'] = {
+                    // We consider the AWS kola tests to be a followup job, so we use `wait: false` here.
+                    build job: 'kola-aws', wait: false, parameters: [
+                        string(name: 'STREAM', value: params.STREAM),
+                        string(name: 'VERSION', value: newBuildID),
+                        string(name: 'S3_STREAM_DIR', value: s3_stream_dir),
+                        string(name: 'ARCH', value: basearch),
+                        string(name: 'SRC_CONFIG_COMMIT', value: src_config_commit)
+                    ]
+                }
+              // XXX: This is failing right now. Disable until the New
+              // Year when someone can dig into the problem.
+              //parallelruns['Kola:Kubernetes'] = {
+              //    // We consider the Kubernetes kola tests to be a followup job, so we use `wait: false` here.
+              //    build job: 'kola-kubernetes', wait: false, parameters: [
+              //        string(name: 'STREAM', value: params.STREAM),
+              //        string(name: 'VERSION', value: newBuildID),
+              //        string(name: 'S3_STREAM_DIR', value: s3_stream_dir),
+              //        string(name: 'SRC_CONFIG_COMMIT', value: src_config_commit)
+              //    ]
+              //}
             }
-          // XXX: This is failing right now. Disable until the New
-          // Year when someone can dig into the problem.
-          //parallelruns['Kola:Kubernetes'] = {
-          //    // We consider the Kubernetes kola tests to be a followup job, so we use `wait: false` here.
-          //    build job: 'kola-kubernetes', wait: false, parameters: [
-          //        string(name: 'STREAM', value: params.STREAM),
-          //        string(name: 'VERSION', value: newBuildID),
-          //        string(name: 'S3_STREAM_DIR', value: s3_stream_dir),
-          //        string(name: 'SRC_CONFIG_COMMIT', value: src_config_commit)
-          //    ]
-          //}
-        }
-        if (!params.MINIMAL && uploading &&
```

Comment:
Don't we still need those conditionals in this and following if-statements?

Reply:
I converted it. Now all of these cloud test kickoff stanzas are nested inside a toplevel `if (!params.MINIMAL && uploading) {`.

Reply:
If you ignore whitespace changes in your diff it might be harder to see this.